components.tsx

```diff
@@ -77,36 +94,45 @@ const styles = StyleSheet.create({
   bottomNavigation: {
     flexDirection: "row",
     alignItems: "center",
-    marginBottom: 24
+    marginBottom: 32
   }
 });
 
-const PrimaryButton = (props: {
+type FontWeightVariants = "SemiBold" | "Medium" | "Bold";
+
+const AppText = (props: {
+  children: React.ReactNode;
+  style?: TextStyle | TextStyle[];
+  fontWeight?: FontWeightVariants;
+}) => {
+  const fontType = !props.fontWeight ? "Regular" : props.fontWeight;
+  return (
+    <Text style={[{ fontFamily: `Montserrat-${fontType}` }, props.style]}>
```

Comment:
What happens if you do:
```
<AppText style={[{ color: 'red' }]} />
```
note that I pass an array into the style prop

Reply:
it works! 
![IMG_5490](https://user-images.githubusercontent.com/13867964/77638435-71c81f80-6f57-11ea-9181-9dda5d5d6861.png)

Reply:
What was the concern here? That I didn't spread the props?

Reply:
The concern was regarding React Native flattening a nested array with styles, imagine trying to use this array of styles:
```[{ color: 'red' }, [{ fontFamily: 'Open Sans'}, [{ fontSize: 16 }]]]```
I was worried that it wouldn't have been implemented